articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,455 @@
+
+single_qubit_result = MeasureSuperposition.simulate()
+single_qubit_resources = MeasureSuperposition.estimate_resources()
+
+multi_qubit_result = MeasureSuperpositionArray.simulate(n=4)
+multi_qubit_resources = MeasureSuperpositionArray.estimate_resources(n=4)
+
+print('Single qubit:')
+print(single_qubit_result)
+print(single_qubit_resources)
+
+print('\nMultiple qubits:')
+print(multi_qubit_result)
+print(multi_qubit_resources)
+```
+
+results in an output like the following:
+
+```python
+Single qubit:
+1
+{'CNOT': 0, 'QubitClifford': 1, 'R': 0, 'Measure': 1, 'T': 0, 'Depth': 0, 'Width': 1, 'BorrowedWidth': 0}
+
+Multiple qubits:
+[1, 1, 0, 0]
+{'CNOT': 0, 'QubitClifford': 4, 'R': 0, 'Measure': 8, 'T': 0, 'Depth': 0, 'Width': 4, 'BorrowedWidth': 0}
+```
+
+### [C#](#tab/tabid-csharp)
+
+A C# host program has multiple components, and works very closely with some components of the QDK, such as the simulators, which are also built on the .NET language.
+
+The Q# compiler works here by generating an equivalently named C# namespace from the Q# namespace in our Q# file.
+It further generates an equivalently named C# class for each of the Q# callables or types defined therein.
+
+First, we make any classes used in our host program available with `using` statements:
+
+```csharp
+using System;
+using Microsoft.Quantum.Simulation.Simulators;    // contains the target machines, e.g. QuantumSimulator
+using NamespaceName;                              // make the Q# namespace available
+```
+
+Next, we declare our C# namespace, a few other bits and pieces (see the full code block below), and then any classical programming we would like (e.g. computing arguments for the Q# operations).
+The latter isn't necessary in our case, but an example of such usage can be found at the  [.NET interoperability sample](https://github.com/microsoft/Quantum/tree/master/samples/interoperability/dotnet).
+
+#### Target machines
+
+Getting back to Q#, we must create an instance of whatever target machine we will execute our operations on.
+
+```csharp
+            using var sim = new QuantumSimulator();
+```
+
+Using other target machines is as simple as instantiating a different one, although processing the returns could require different action.
+For an example, you can check out how to use the [`ResourceEstimator` with C#](xref:microsoft.quantum.machines.resources-estimator).
+
+Each C# class generated from the Q# operations have a `Run` method, the first argument of which must be the target machine instance.
+So, to run `MeasureSuperposition` on the `QuantumSimulator`, we use `MeasureSuperposition.Run(sim)`.
+The returned results can then be assigned to variables in C# by fetching the `Result` property:
+
+```csharp
+            var single_qubit_result = MeasureSuperposition.Run(sim).Result;
+```
+
+> [!NOTE]
+> If the Q# operation does not have any returns (i.e. has return type `Unit`), use the `.Wait()` method call instead of `.Result`.
```

Comment:
We've started trying to use the `await` keyword here, since that works both on methods without a return type (i.e.: `void`) and on methods with a return type. That also has the advantage of looking more like Python's or JavaScript's `async`/`await`.